src/middleware/index.js

```diff
@@ -80,6 +81,18 @@ module.exports = function(app) {
 
 	app.use(middleware.processRender);
 	auth.initialize(app, middleware);
+	
+	app.use(function(req, res, next) {
+        	if (plugins.hasListeners('action:request.earlyProcessing')) {
```

Comment:
Hook should be a filter, actally. The way it's written now, two separate plugins could both attach to this hook, and both could execute `res.send`, causing a `Headers already sent` exception and crashing NodeBB.

If you turn this to a filter, you can get rid of the `hasListeners` call as well.